firedantic/_async/model.py

```diff
@@ -38,6 +49,11 @@
     op.NOT_IN,
 }
 
+IndexField = NamedTuple("IndexField", [("name", str), ("order", str)])
+IndexDef = NamedTuple(
+    "IndexDef", [("query_scope", str), ("fields", Tuple[IndexField, ...])]
```

Comment:
Generally I'd prefer `IndexDefinition`, but no big deal, feel free to resolve as-is.